### PR TITLE
Enhancement/#127 creation of contributor license agreement

### PR DESCRIPTION
**Pull request by ChristophBaierATB**
closes coatrack#127

### CODE REVIEW

File: cla/signing-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+## Structure of the 'cla' folder
+
+For transparency reasons, we will briefly explain here how our CLA system works. Harmony provides [templates](http://harmonyagreements.org/agreements.html) for CLA's in ODT format which are stored in the [original](https://github.com/coatrack/cla/source/original) folder. Based on that we made some customizations fitting our requirements which are stored in the [customized](https://github.com/coatrack/cla/source/customized) folder. With LibreOffice, for example, you can easily see the differences between the original and our customized version by comparing the two ODT files. The customized CLA ODT's are converted to PDF's and stored in the [latest](https://github.com/coatrack/cla/latest) folder. These PDF files are the ones that have to be signed/hashed in order to be allowed to contribute to CoatRack.
```
Comment (BrunoSilvaATB) [inline cla/signing-tutorial.md:9]: Based on that,

File: cla/signing-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+## License
+
+By contributing, you agree that your contributions will be licensed under Apache License 2.0. When you submit code changes, your submissions are understood to be under the same Apache License 2.0 that covers the project.
```
Comment (BrunoSilvaATB) [inline cla/signing-tutorial.md:15]: Here maybe should be something like "by signing this  CLA" instead of by contributing.
Because here, from what I understood, you are just talking about "the CLA" itself and not the contribution

File: cla/signing-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+## References
+
+This document was derived from the [CONTRIBUTING.md](https://github.com/adam-p/markdown-here/blob/master/CONTRIBUTING.md) from the [adam-p/markdown-here](https://github.com/adam-p/markdown-here) project. The content of the [signing-template.md](https://github.com/coatrack/signing-template.md) was derived from [adam-p's agreement file](https://github.com/adam-p/markdown-here/blob/master/contributors/adam-p.md).
```
Comment (BrunoSilvaATB) [inline cla/signing-tutorial.md:21]: Here the word "from" is repeated two times in sequence, which could be confusing to the users, maybe rephrase it

File: contribution-and-developer-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Hello, dear prospective contributor. You can support CoatRack in many ways:
+
+- Report bugs using GitHub's issue tracker.
+- Discuss the current state of the code in open Pull Requests.
+- Submit a bugfix.
+- Propose new features and improvement ideas.
+- Improve the documentation.
+- Clean up code.
+- Add tests.
+- Help us to fulfill the [FIWARE requirements](https://fiware-requirements.readthedocs.io/en/latest/).
```
Comment (BrunoSilvaATB) [inline contribution-and-developer-guidelines.md:18]: I think this wording here could be misleading.
The contributor will fill up also the FIWARE requirements directly?

File: contribution-and-developer-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+
+## Conventions
+
+#### Applying Best Practices
+
+The establishment of best practices in CoatRack and compliance of these is important to us. We put great effort to our review process to steadily improve the quality of our code base and we expect the same willingness from you. Here are a few practices that should always be considered: 
+
+* Code is written for readers and should be quickly understood by them. Use descriptive names, a clear architecture, short functions and small, cohesive classes. Add Java Docs if necessary.
+* Use a consistent coding style that goes along with the already existing code. Please do not change the formatting of exiting code that is not relevant to the issue you are working on.
```
Comment (BrunoSilvaATB) [inline contribution-and-developer-guidelines.md:33]: existing instead of exiting

File: contribution-and-developer-guidelines.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+* Spaces should be used for indentation, not tabs.
+* Newly added code should be delivered with automated tests, for example unit tests or integration tests. Ensure to have a high code coverage.
+
+
+
+#### Workflow
+
+We use GitHub to host code, track issues and feature requests, as well as accept pull requests. We use the following workflow:
+
+1. Fork the repo. Of course, this step must be performed only once. All others are run again for each issue.
```
Comment (BrunoSilvaATB) [inline contribution-and-developer-guidelines.md:43]: Should we also refer here that the user/contributor should disable the worflow that tries to push the code of the master branch to dev.

File: contribution-and-developer-guidelines.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+5. Before creating a pull request, ensure that
+   1. all tests are are passing,
+   2. the comment of the pull request contains at least this line: 
+      * `closes #<issue-number>` .
+
+6. At least two active maintainers of the projects need to review your code before it is allowed to be merged.
+
+
+
+#### How to Make a Bug Report
```
Comment (BrunoSilvaATB) [inline contribution-and-developer-guidelines.md:67]: Here should we have an example of a Bug report that follow the approach described.
I think it might be easier for the contributor to understand

File: contribution-and-developer-guidelines.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+* Add a quick descriptive summary and/or background information.
+* Tell us what behavior of the application you expected and what you actually encountered.
+* Screenshots and log files are very helpful for us.
+* You should be able to reproduce the bug and enable us to do the same. Write down every single step that leads to the bug.
+* If you have a guess why this bug might be happening or why you think, it didn't work, then tell us.
```
Comment (BrunoSilvaATB) [inline contribution-and-developer-guidelines.md:73]: why you think it didn't work, so I think that the comma is not correct there

**Review by BrunoSilvaATB — CHANGES_REQUESTED** [review]
So @ChristophBaierATB I reviewed the text and made some suggestions. 
Let us discuss this on our TelCo in the afternoon

**Comment by BrunoSilvaATB** [comment]
I also checked the workflow, and the "checks failed" seems to be because of the tests you wrote.
All of them test of the MetricsAggregationRepository class

image

File: cla/signing-tutorial.md (hunk 1 above)
Comment (ChristophBaierATB) [reply cla/signing-tutorial.md]: Done.

File: cla/signing-tutorial.md (hunk 3 above)
Comment (ChristophBaierATB) [reply cla/signing-tutorial.md]: Done.

File: contribution-and-developer-guidelines.md (hunk 8 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done.

File: cla/signing-tutorial.md (hunk 2 above)
Comment (BrunoSilvaATB) [reply cla/signing-tutorial.md:15]: reminder: Please check standard (check template used to create the CLA)

File: contribution-and-developer-guidelines.md (hunk 7 above)
Comment (BrunoSilvaATB) [reply contribution-and-developer-guidelines.md:67]: Create follow up issue about creating a issue report template

File: cla/signing-template.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<yyyy-mm-dd>
+
+I hereby agree to the terms of the "CoatRack <Entity or Individual> Contributor License Agreement", with SHA-256 checksum <sha-256-checksum>.
+
+I furthermore declare that I am authorized and able to make this agreement and sign this declaration.
+
+Signed,
+
+<firstname> <lastname> https://github.com/<github-username>
```
Comment (gr-hovest-atb) [inline cla/signing-template.md:9]: I think we should not ask people to put their full name here, the Github username should be sufficient as it clearly maps the contribution to a Github account. 

The personal information like name, address, should be sent to us e.g. via Email instead.

File: cla/signing-tutorial.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# CoatRack CLA Signing Tutorial
```
Comment (gr-hovest-atb) [inline cla/signing-tutorial.md:1]: "Tutorial" is not a good name for this, I would call it "instructions". In general I am wondering if this shouldn't be part of the contribution guidelines file? All information in one page seems to be helpful for prospective contributors.

File: cla/signing-tutorial.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+When submitting a pull request for the first time, you will need to agree to the contributor license agreement (CLA). For example single, private persons must agree on the [CLA for individuals](https://github.com/coatrack/cla/latest/individual-cla.pdf) and organizations must agree on the [CLA for entities](https://github.com/coatrack/cla/latest/entity-cla.pdf). To do this, copy, rename and move [signing-template.md](https://github.com/coatrack/cla/signing-template.md) to `/cla/contributors/<your-github-username>.md`. Fill in the missing information, including the SHA-256 checksum of the corresponding CLA just mentioned, and add this file to your first pull request.
```
Comment (gr-hovest-atb) [inline cla/signing-tutorial.md:3]:
```suggestion
When submitting a pull request for the first time, you will need to agree to the contributor license agreement (CLA). 

* Individual private persons must agree on the [CLA for individuals](https://github.com/coatrack/cla/latest/individual-cla.pdf) and 
* Organizations must agree on the [CLA for entities](https://github.com/coatrack/cla/latest/entity-cla.pdf). 

If you agree to the respective CLA, you have to create a pull request including a file that officially and transparently documents your intention to sign the CLA:

 1. Please copy, rename and move [signing-template.md](https://github.com/coatrack/cla/signing-template.md) to `/cla/contributors/<your-github-username>.md`. 
 2. Fill in the missing information, including the SHA-256 checksum of the CLA PDF file you want to sign, and add this file to your first pull request. The checksum clearly documents the version of the CLA that you agree to.
 3. After you created the pull request, please send an email to `info_(at)_coatrack_dot_eu` to inform us about the pull request. 
 
By merging your pull request, the the information that you agreed to the CLA will be part of the official public code base of CoatRack.
 
```

File: cla/signing-tutorial.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+## Structure of the 'cla' folder
+
+For transparency reasons, we will briefly explain here how our CLA system works. Harmony provides [templates](http://harmonyagreements.org/agreements.html) for CLA's in ODT format which are stored in the [original](https://github.com/coatrack/cla/source/original) folder. Based on that, we made some customizations fitting our requirements which are stored in the [customized](https://github.com/coatrack/cla/source/customized) folder. With LibreOffice, for example, you can easily see the differences between the original and our customized version by comparing the two ODT files. The customized CLA ODT's are converted to PDF's and stored in the [latest](https://github.com/coatrack/cla/latest) folder. These PDF files are the ones that have to be signed/hashed in order to be allowed to contribute to CoatRack.
+
+
+
+## License
```
Comment (gr-hovest-atb) [inline cla/signing-tutorial.md:13]: I would put this information _before_ the signing instructions.

File: cla/signing-tutorial.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+
+
+## Structure of the 'cla' folder
+
+For transparency reasons, we will briefly explain here how our CLA system works. Harmony provides [templates](http://harmonyagreements.org/agreements.html) for CLA's in ODT format which are stored in the [original](https://github.com/coatrack/cla/source/original) folder. Based on that, we made some customizations fitting our requirements which are stored in the [customized](https://github.com/coatrack/cla/source/customized) folder. With LibreOffice, for example, you can easily see the differences between the original and our customized version by comparing the two ODT files. The customized CLA ODT's are converted to PDF's and stored in the [latest](https://github.com/coatrack/cla/latest) folder. These PDF files are the ones that have to be signed/hashed in order to be allowed to contribute to CoatRack.
```
Comment (gr-hovest-atb) [inline cla/signing-tutorial.md:9]:
```suggestion
For transparency reasons, we will briefly explain here how our CLA was created and how the signing process works. 
Harmony provides [templates](http://harmonyagreements.org/agreements.html) for CLA's in ODT format which are stored in the [original](https://github.com/coatrack/cla/source/original) folder. Based on those templates we made custom versions, adapted to our CoatRack requirements, which are stored in the [customized](https://github.com/coatrack/cla/source/customized) folder. If you want to see the differences between the original and our customized version, you can easily compare the two ODT files (e.g. using LibreOffice). 

The customized CLA ODT's are converted to PDF's and stored in the [latest](https://github.com/coatrack/cla/latest) folder. These PDF files are the ones that have to be signed/hashed in order to be allowed to contribute to CoatRack.
```

File: contribution-and-developer-guidelines.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+#### Contact
+
+If you have questions, remarks or you want to get involved in the project, please contact ```info@coatrack.eu```.
+
+
+
+#### References
+
+This document was derived from [briandk/CONTRIBUTING.md](https://gist.github.com/briandk/3d2e8b3ec8daf5a27a62).
```
Comment (gr-hovest-atb) [inline contribution-and-developer-guidelines.md:102]: On first glance, that linked file looks very different from this one, so maybe use a word like "inspired" or similar, rather than "derived"?

File: contribution-and-developer-guidelines.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
+# CoatRack Contribution and Developer Guidelines
```
Comment (gr-hovest-atb) [inline contribution-and-developer-guidelines.md:1]: As developers are a sub-group of contributors, I think the following should be sufficient:
```suggestion
# CoatRack Contributor Guidelines
```

Besides, I would suggest to rename this file as `CONTRIBUTING.md`, as that is the filename recommended in the FIWARE requirements. I would also suggest to include the "signing instructions" in this file, if there is no obvious reason to do otherwise.

File: contribution-and-developer-guidelines.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+
+
+
+## Introduction
+
+#### Possibilities to Contribute
+
+Hello, dear prospective contributor. You can support CoatRack in many ways:
+
+- Report bugs using GitHub's issue tracker.
```
Comment (gr-hovest-atb) [inline contribution-and-developer-guidelines.md:11]:
```suggestion
- Report bugs using CoatRack's Github issue tracker.
```

File: contribution-and-developer-guidelines.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+- Discuss the current state of the code in open Pull Requests.
```
Comment (gr-hovest-atb) [inline contribution-and-developer-guidelines.md:12]:
```suggestion
- Discuss suggested code changes in open Pull Requests.
```

File: contribution-and-developer-guidelines.md (hunk 4 above)
Comment (gr-hovest-atb) [reply contribution-and-developer-guidelines.md:18]: As we have already fullfilled many, I would say: 
```suggestion
- Help us to fulfill all [FIWARE requirements](https://fiware-requirements.readthedocs.io/en/latest/).
```

File: contribution-and-developer-guidelines.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+- Submit a bugfix.
+- Propose new features and improvement ideas.
+- Improve the documentation.
+- Clean up code.
+- Add tests.
+- Help us to fulfill the [FIWARE requirements](https://fiware-requirements.readthedocs.io/en/latest/).
+
+
+
+Note: If you want to contribute to CoatRack you need to sign the corresponding Contributors License Agreement (CLA) previously. Signing instructions can be found in [signing-tutorial.md](https://github.com/coatrack/signing-tutorial.md).
```
Comment (gr-hovest-atb) [inline contribution-and-developer-guidelines.md:22]:
```suggestion
Note: If you want to contribute to CoatRack you need to sign the corresponding Contributors License Agreement (CLA) beforehand. Signing instructions can be found in [signing-tutorial.md](https://github.com/coatrack/signing-tutorial.md). 
```

File: contribution-and-developer-guidelines.md [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+
+
+
+## Conventions
+
+#### Applying Best Practices
+
+The establishment of best practices in CoatRack and compliance of these is important to us. We put great effort to our review process to steadily improve the quality of our code base and we expect the same willingness from you. Here are a few practices that should always be considered: 
+
+* Code is written for readers and should be quickly understood by them. Use descriptive names, a clear architecture, short functions and small, cohesive classes. Add Java Docs if necessary.
```
Comment (gr-hovest-atb) [inline contribution-and-developer-guidelines.md:32]:
```suggestion
* Code is written for readers and should be quickly understood by them. Use descriptive names, a clear architecture, short functions and small, cohesive classes. Add comments and Java Doc if required by readers to understand the code.
```

File: contribution-and-developer-guidelines.md [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+We use GitHub to host code, track issues and feature requests, as well as accept pull requests. We use the following workflow:
+
+1. Fork the repo. Of course, this step must be performed only once. All others are run again for each issue.
+
+2. You can either work on an existing issue or you have to create a descriptive issue before you can make code changes. 
+
+3. Create a branch from `master` for the issue you are working on. 
+
+   * Use this naming scheme: 
+     * `<commit-type>/#<issue-number>-<short-description>`.
```
Comment (gr-hovest-atb) [inline contribution-and-developer-guidelines.md:50]: we should discuss if we really need the hash sign in the branch name (is a little bit annoying as it has to be escaped on the commandline).

**Review by gr-hovest-atb — CHANGES_REQUESTED** [review]
I have rewritten some paragraphs and suggested some wording changes. I would also suggest to put all instructions in one single CONTRIBUTING.md file.

File: contribution-and-developer-guidelines.md (hunk 6 above)
Comment (gr-hovest-atb) [reply contribution-and-developer-guidelines.md:43]: If that is still required, we should add it here.

File: contribution-and-developer-guidelines.md (hunk 20 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md:50]: I don't care about the presence of a hash sign. If it causes unnecessary problems, we should leave it out.

File: contribution-and-developer-guidelines.md (hunk 5 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done.

File: contribution-and-developer-guidelines.md (hunk 6 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done. The above mentioned phrase suggested by @BrunoSilvaATB was added.

File: contribution-and-developer-guidelines.md (hunk 7 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md:67]: Done, see coatrack#165.

File: cla/signing-template.md (hunk 9 above)
Comment (ChristophBaierATB) [reply cla/signing-template.md]: Done. The fields 'firstname' and 'lastname' were removed.

File: cla/signing-tutorial.md (hunk 10 above)
Comment (ChristophBaierATB) [reply cla/signing-tutorial.md]: Done. I renamed the 'signing tutorial' to 'signing instructions' and merged them with the guidelines in a CONTRIBUTION.md file.

File: cla/signing-tutorial.md (hunk 11 above)
Comment (ChristophBaierATB) [reply cla/signing-tutorial.md]: Done as suggested.

File: cla/signing-tutorial.md (hunk 12 above)
Comment (ChristophBaierATB) [reply cla/signing-tutorial.md]: Done.

File: contribution-and-developer-guidelines.md (hunk 14 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done. I replaced 'derived' with 'inspired'.

File: contribution-and-developer-guidelines.md (hunk 15 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done. I merged the guidelines and the signing instructions into a CONTRIBUTING.md file.

File: contribution-and-developer-guidelines.md (hunk 16 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done.

File: contribution-and-developer-guidelines.md (hunk 17 above)
Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done.

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+
+
+## Introduction
+
+#### Possibilities to Contribute
+
+Hello, dear prospective contributor. You can support CoatRack in many ways:
+
+- Report bugs using CoatRack's GitHub issue tracker.
+- Discuss suggested code changes in open Pull Requests.
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:12]:
```suggestion
- Discuss suggested code changes in open pull requests.
```

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+- Submit a bugfix.
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:13]: just to be consistent, all others are also plural:
```suggestion
- Submit bugfixes.
```

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+- Propose new features and improvement ideas.
+- Improve the documentation.
+- Clean up code.
+- Add tests.
+- Help us to fulfill the [FIWARE requirements](https://fiware-requirements.readthedocs.io/en/latest/).
+
+
+
+Note: If you want to contribute to CoatRack you need to sign the corresponding Contributors License Agreement (CLA) beforehand. Signing instructions can be found in In the corresponding section of this document.
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:22]:
```suggestion
Note: If you want to contribute to CoatRack, you need to sign the corresponding Contributors License Agreement (CLA) beforehand. Signing instructions can be found in In the corresponding section of this document.
```

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,137 @@
...
+We use GitHub to host code, track issues and feature requests, as well as accept pull requests. We use the following workflow:
+
+1. Fork the repo. Make sure to disable the GitHub action that tries to push the code of the master branch to our 'dev' deployment environment. Of course, this step must be performed only once. All other steps are run again for each issue.
+
+2. You can either work on an existing issue or you have to create a descriptive issue before you can make code changes. 
+
+3. Create a branch from `master` for the issue you are working on. 
+
+   * Use this naming scheme: 
+     * `<commit-type>/#<issue-number>-<short-description>`.
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:50]: Instead of "commit type" I would call it "issue type", as one branch/PR will include many commits.

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -0,0 +1,137 @@
...
+   2. the comment of the pull request contains at least this line: 
+      * `closes #<issue-number>` .
+
+6. At least two active maintainers of the projects need to review your code before it is allowed to be merged.
+
+
+
+#### How to Make a Bug Report
+
+* Add a quick descriptive summary and/or background information.
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:69]: Not sure what "quick" means here. Remove the word?

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,137 @@
...
+
+
+
+#### Recommended Terminology
+
+To avoid misunderstandings, we have decided to provide a uniform terminology recommendation for the most important concepts in CoatRack. Try to avoid terms like 'server' and 'client' because the Gateway can be both in different contexts and therefore leads to confusion. As this recommendation did not exist from the very beginning of the project, you may encounter different wordings for the same concept. Here is an overview:
+
+| Recommended Wording      | Also Used Wording              | Description                                                  |
+| ------------------------ | ------------------------------ | ------------------------------------------------------------ |
+| CoatRack Web Application | (CoatRack) Admin               | This is the central CoatRack component which can be reached at [coatrack.eu](https://coatrack.eu/). It offers a web UI to define services as well as creating gateways and API keys. |
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:83]:
```suggestion
| CoatRack Web Application | (CoatRack) Admin, ygg-admin               | This is the central CoatRack component which can be reached at [coatrack.eu](https://coatrack.eu/). It offers a web UI to define services as well as creating gateways and API keys. |
```

File: CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,137 @@
...
+| Gateway                  | Proxy                          | It grants access to the specific services if a Service User sends a valid API key to the gateway. |
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:84]:
```suggestion
| Gateway                  | Proxy, ygg-proxy                          | It grants access to the specific services if a Service User sends a valid API key to the gateway. |
```

File: CONTRIBUTING.md [hunk 28]
```diff
@@ -0,0 +1,137 @@
...
+
+By contributing, you agree that your contributions will be licensed under Apache License 2.0. When you submit code changes, your submissions are understood to be under the same Apache License 2.0 that covers the project.
+
+
+
+### Signing Instructions
+
+When submitting a pull request for the first time, you will need to agree to the contributor license agreement (CLA). 
+
+* Individual private persons must agree on the [CLA for individuals](https://github.com/coatrack/cla/latest/individual-cla.pdf) and
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:104]:
```suggestion
* Individual private persons must agree to the [CLA for individuals](https://github.com/coatrack/coatrack/blob/master/cla/latest/individual-cla.pdf) and 
```

File: CONTRIBUTING.md [hunk 29]
```diff
@@ -0,0 +1,137 @@
...
+
+
+
+### Signing Instructions
+
+When submitting a pull request for the first time, you will need to agree to the contributor license agreement (CLA). 
+
+* Individual private persons must agree on the [CLA for individuals](https://github.com/coatrack/cla/latest/individual-cla.pdf) and 
+
+* Organizations must agree on the [CLA for entities](https://github.com/coatrack/cla/latest/entity-cla.pdf).
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:106]:
```suggestion
* Organizations must agree to the [CLA for entities](https://github.com/coatrack/coatrack/blob/master/cla/latest/entity-cla.pdf). 
```

File: CONTRIBUTING.md [hunk 30]
```diff
@@ -0,0 +1,137 @@
...
+
+When submitting a pull request for the first time, you will need to agree to the contributor license agreement (CLA). 
+
+* Individual private persons must agree on the [CLA for individuals](https://github.com/coatrack/cla/latest/individual-cla.pdf) and 
+
+* Organizations must agree on the [CLA for entities](https://github.com/coatrack/cla/latest/entity-cla.pdf). 
+
+If you agree to the respective CLA, you have to create a pull request including a file that officially and transparently documents your intention to sign the CLA:
+
+1. Copy, rename and move [signing-template.md](https://github.com/coatrack/cla/signing-template.md) to `/cla/contributors/<your-github-username>.md`.
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:110]:
```suggestion
1. Copy, rename and move [signing-template.md](https://github.com/coatrack/coatrack/blob/master/cla/signing-template.md) to `/cla/contributors/<your-github-username>.md`. 
```

File: CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,137 @@
...
+
+* Organizations must agree on the [CLA for entities](https://github.com/coatrack/cla/latest/entity-cla.pdf). 
+
+If you agree to the respective CLA, you have to create a pull request including a file that officially and transparently documents your intention to sign the CLA:
+
+1. Copy, rename and move [signing-template.md](https://github.com/coatrack/cla/signing-template.md) to `/cla/contributors/<your-github-username>.md`. 
+2. Fill in the missing information, including the SHA-256 checksum of the CLA PDF file you want to sign, and add this file to your first pull request. The checksum clearly documents the version of the CLA that you agree to.
+3. After you created the pull request, please send an email to `info@coatrack.eu` to inform us about the pull request. 
+
+By merging your pull request, the information that you agreed to the CLA will be part of the official public code base of CoatRack.
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:114]:
```suggestion
By merging your pull request, the information that you agreed to the CLA will become part of the official public code base of CoatRack.
```

File: CONTRIBUTING.md [hunk 32]
```diff
@@ -0,0 +1,137 @@
...
+
+
+
+### Structure of the 'cla' folder
+
+For transparency reasons, we will briefly explain here how our CLA was created and how the signing process works. Harmony provides [templates](http://harmonyagreements.org/agreements.html) for CLA's in ODT format which are stored in the [original](https://github.com/coatrack/cla/source/original) folder. Based on those templates we made custom versions, adapted to our CoatRack requirements, which are stored in the [customized](https://github.com/coatrack/cla/source/customized) folder. If you want to see the differences between the original and our customized version, you can easily compare the two ODT files (e.g. by using the document comparison feature of LibreOffice).
```
Comment (gr-hovest-atb) [inline CONTRIBUTING.md:120]:
```suggestion
For transparency reasons, we will briefly explain here how our CLA was created and how the signing process works. Harmony provides [templates](http://harmonyagreements.org/agreements.html) for CLA's in ODT format which are stored in the [original](https://github.com/coatrack/coatrack/tree/master/cla/source/original) folder. Based on those templates we made custom versions, adapted to our CoatRack requirements, which are stored in the [customized](https://github.com/coatrack/coatrack/tree/master/cla/source/customized) folder. If you want to see the differences between the original and our customized version, you can easily compare the two ODT files (e.g. by using the document comparison feature of LibreOffice). 
```

**Review by gr-hovest-atb — CHANGES_REQUESTED** [review]
I have added a few additional improvement suggestions. Afterwards I would suggest to create a draft PR to the main repo master (I think that some details still need to be discussed, therefore it should be a draft).

The customized ODT and PDF files should all get a "CoatRack-" prefix in their name. Some specific details we should discuss in our next call:
- we should explicitly say something about "contributors who do not own the copyright for the entire contribution", e.g. "then please do not contribute"
- Country of jurisdiction could also be Netherlands (to be discussed)
- Maybe we should add Corizon and ATB company data (name, address, etc.) into the CLA as "Us"

File: contribution-and-developer-guidelines.md (hunk 15 above)
Comment (gr-hovest-atb) [reply contribution-and-developer-guidelines.md]: Just a minor note: the ODT/PDF files still refer to "signing-template.md", should be changed.

File: contribution-and-developer-guidelines.md (hunk 20 above)
Comment (gr-hovest-atb) [reply contribution-and-developer-guidelines.md]: Then please remove :)

Comment (ChristophBaierATB) [reply contribution-and-developer-guidelines.md]: Done.

File: CONTRIBUTING.md (hunk 21 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 22 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 23 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 24 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 25 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done by removal of the word.

File: CONTRIBUTING.md (hunk 26 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 27 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md [hunk 33]
```diff
@@ -0,0 +1,137 @@
...
+
+#### Contact
+
+If you have questions, remarks or you want to get involved in the project, please contact ```info@coatrack.eu```.
+
+
+
+#### References
+
+This document was inspired from the works of [briandk](https://gist.github.com/briandk/3d2e8b3ec8daf5a27a62) and [adam-p](https://github.com/adam-p/markdown-here/blob/master/CONTRIBUTING.md). The content of the [signing-template.md](https://github.com/coatrack/signing-template.md) was derived from [adam-p's agreement file](https://github.com/adam-p/markdown-here/blob/master/contributors/adam-p.md). The signing method was adopted from the [medium/opensource](https://github.com/medium/opensource) project.
```
[the remark this answers is not in the document]
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 31 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 30 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 29 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 28 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 27 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 32 above)
Comment (ChristophBaierATB) [reply CONTRIBUTING.md]: Done.
